### ISSUE TITLE
Events emitted from subcontract in contract->subcontract calls are decoded as events from the caller contract

### ISSUE THREAD

**Issue by vminkov** [polkadot-js/api#3330]
I will add in the comments later a repo with a simple example, but the title should be enough to get the idea - all emitted events currently are treated as events emitted from the top caller smart contract, so events from subcontract calls are wrongly decoded or fail to decode because the framework looks them up in the top caller contract ABI event index

**Comment by jacogr** [comment]
A reproduction with wasm/json/contract files would be great. However, can see how that happens, hence the request for play-samples, to see if it is properly solvable.

**Comment by vminkov** [comment]
Here is a complete example, please note that contractEvents will be undefined unless fixed ( https://github.com/polkadot-js/api/pull/3324 )

The breaking code is in the Mocha files test/test.js

https://github.com/vminkov/polkadotjs-events-issue

**Comment by jacogr** [comment]
Yes, looking at the code quickly - there is absolutely no link between the contract and the sub ABI. So possibly need to explicitly pass all sub-ABIs through to a specific instance.

**Comment by valentinfernandez1** [comment]
Outdated issue
